Add many descriptions to the `Window` class

SirQuartz opened the pull request:
Added several missing descriptions for various functions and properties. I don't fully understand what every function of this node does yet, but I think I'm on the right track with these additions after some testing. Plus it's better than nothing. I intend to improve this wherever possible.

If you've any feedback, recommendations, and/or insights I'd be glad to hear it.

aaronfranke committed:
Flip 2D icon to match Godot's handedness

Calinou committed:
Increase the default AudioStreamPlayer3D unit size to 10

This makes it easier to hear sound while setting up the node.

Since this changes the default value, this may break existing projects
slightly.

This also tweaks the Unit Size editor property hint for better usability.

See discussion in #25468.

aaronfranke committed:
Update documentation for Transform3D

aaronfranke committed:
Rename Transform to Transform3D in GDNative

aaronfranke committed:
Rename Transform to Transform3D in core

madmiraal committed:
Rename Node3D's property translation to position

aaronfranke committed:
Allow clamping vectors and colors

aaronfranke committed:
Rename Vector2 clamped to limit_length and add limit_length to Vector3

aaronfranke committed:
Rename files and the exposed name for Transform3D

aaronfranke committed:
Disable 3D-only modules when 3D is disabled

Calinou committed:
Rename the `.shader` file extension to `.gdshader`

This lets third-party software recognize Godot shaders more easily,
without relying on guesswork since the `.shader` extension is generic.

aaronfranke committed:
Add MOUSE_MODE_CONFINED_HIDDEN

Co-authored-by: Hugo Locurcio <hugo.locurcio@hugo.pro>

aaronfranke committed:
Rename tabs in the project manager

aaronfranke committed:
Rename Variant TRANSFORM to TRANSFORM3D

Also _transform to _transform3d

Calinou committed:
Rename File's `endian_swap` to `big_endian`

This new name is more consistent with ResourceSaver and StreamPeer.

YuriSizov committed:
Improve the UI/UX of the Export Template Manager dialog

sarchar committed:
Fix DNS resolve mutex locks

This fixes #49261, which was happening because of a deadlock in the resolver mutex.  There was leftover old mutex code and it's all be converted to new MutexLock class now.

YuriSizov committed:
Replace get_settings_dir with get_config_dir when fetching configuration paths

KoBeWi committed:
Improve sort_custom() description

aaronfranke committed:
Rename Animation TYPE_TRANSFORM to TYPE_TRANSFORM3D

aaronfranke committed:
Rename EditorPropertyTransform to EditorPropertyTransform3D

akien-mga committed:
doc: Sync classref after merge of #47336

BastiaanOlij committed:
Rename GODOT_VARIANT_TYPE_TRANSFORM to GODOT_VARIANT_TYPE_TRANSFORM3D

foxydevloper commented:
The descriptions of the properties should be changed to not start with "sets" or "returns", since properties can both be set and return. You can look through the docs to see how other other classes' properties' descriptions are written.
For example, instead of "Returns `true` if the window can't be resized.", you could write "If `true`, the window can't be resized." which is more consistent the rest of the documentation.

foxydevloper commented on doc/classes/Window.xml:
```diff
@@ -333,16 +343,21 @@
 		</member>
 		<member name="mode" type="int" setter="set_mode" getter="get_mode" enum="Window.Mode" default="0">
+			Sets the window's current mode (i.e windowed, fullscreen, minimized, etc).
+			[b]Note:[/b] Fullescreen mode is not exclusive fullscreen on Windows and Linux.
```
```suggestion
			[b]Note:[/b] Fullscreen mode is not exclusive fullscreen on Windows and Linux.
```

SirQuartz commented:
> The descriptions of the properties should be changed to not start with "sets" or "returns", since properties can both be set and return. You can look through the docs to see how other classes' properties' descriptions are written.
> For example, instead of "Returns `true` if the window can't be resized.", you could write "If `true`, the window can't be resized." which is more consistent the rest of the documentation.

Thank you! I'll see to fixing that!.

foxydevloper commented:
It seems you also changed the descriptions for the methods, those make sense to start with "Return". I was only talking about the classes' properties. Otherwise the change look good.

SirQuartz commented:
> It seems you also changed the descriptions for the classes' methods, those made sense to start with "Returns". I only suggested changing the classes' properties' descriptions. Otherwise the change look good.

Thanks! I've updated the classes' methods to start "Returns" again!

pycbouh commented:
Please squash your commits. You need to use a desktop Git client for that, you can't do it using GitHub on web.

https://docs.godotengine.org/en/stable/community/contributing/pr_workflow.html

SirQuartz committed:
Added several missing descriptions to the Window class.

SirQuartz commented:
> Please squash your commits. You need to use a desktop Git client for that, you can't do it using GitHub on web.
> 
> https://docs.godotengine.org/en/stable/community/contributing/pr_workflow.html
> 
> You will have to do this after reviews as well, so it's a good time to practice now.

Ok, I think I've managed to do it! Thank you!

Chaosus commented:
> Please squash your commits.

That deadly advice :D

SirQuartz commented:
> > Please squash your commits.
> 
> That deadly advice :D

Yeah, I'll be honest, I'm not very familiar with Git, and I'm still learning a lot about it, so things got a _little_ messy. Anyway, I've opened a new pull request that should be fine now with all the edits/additions.

Calinou commented:
Superseded by https://github.com/godotengine/godot/pull/49344.

pycbouh commented:
> > Please squash your commits.
> 
> That deadly advice :D

They've done a couple of force-pushes completely fine! But oh well...
